pandas/core/indexes/api.py

```diff
@@ -112,7 +112,7 @@ def _get_combined_index(indexes, intersect=False, sort=False):
     elif intersect:
         index = indexes[0]
         for other in indexes[1:]:
-            index = index.intersection(other, sort=sort)
+            index = index.intersection(other)
```

Comment:
For some concat tests. Basically, this `sort=sort` was only added in the PR that added a sort keyword (so before it was working fine without it), and I assume it was needed because initially it had a default of True. But now we switched it back to False, we don't need it. 
And the reason the tests were failing is because it was passing True, which is not allowed.